utils/evaluate/evaluate.go

```diff
@@ -185,3 +187,49 @@ func Equal(a, b []string) bool {
 	}
 	return true
 }
+
+func defaultFunc(resultValue interface{}) func(interface{}, interface{}) interface{} {
+	return func(_ interface{}, defaultValue interface{}) interface{} {
+		if isNil(resultValue) {
+			return defaultValue
+		}
+		return valueFromPointer(resultValue)
+	}
+}
+
+func isNilFunc(resultValue interface{}) func(interface{}) bool {
+	return func(_ interface{}) bool {
+		return isNil(resultValue)
+	}
+}
+
+// isNil is courtesy of: https://gist.github.com/mangatmodi/06946f937cbff24788fa1d9f94b6b138
+func isNil(in interface{}) (out bool) {
+	if in == nil {
+		out = true
+		return
+	}
+
+	switch reflect.TypeOf(in).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		out = reflect.ValueOf(in).IsNil()
+	}
+
+	return
+}
+
+// valueFromPointer allows pointers to be passed in from the provider, but then extracts the value from
+// the pointer if the pointer is not nil, else returns nil
+func valueFromPointer(in interface{}) (out interface{}) {
```

Comment:
@jessesuen there's a design implementation question here.

Right now, it's coded as `valueFromPointer()` which allows `resultValue` to be:
- a value
- a pointer
- a nil pointer

Although it's already coded, the downsides are:
- The use of the `reflect` package
  - In general it should be used sparingly--only if necessary
  - Although all unit tests for all providers succeed, there might be other data types, functions, etc. that could break the logic. I'm decently comfortable with the `reflect` package logic here, but I'm not a wizard with the `reflect` package--there could be pitfalls I'm missing.
  
Alternatively, I can remove `valueFromPointer` and create `valueOrNil` function--which would reduce the overall functionality and only allow `resultValue` to be one of two datatypes:
- a value
- a nil pointer

The downside with this approach, is the providers won't be allowed to pass in `a pointer` where the value is parsed and evaluated in the condition. If `a pointer` was provided, the evaluation would error because of a datatype mismatch. This would require a developer to redo this work I have here if `a pointer` was required to be passed in as `resultValue` from the provider.

TBH, my vote is to leave it as is. **But**, I wanted to bring this up to everyone so its documented and everyone is made aware.

Reply:
This seems fine. In the end, I don't think this matters much because responses from providers will likely not have pointer representations of their payload responses. So I think this is an implementation detail that doesn't otherwise affect the overall approach. We can stick with what you have.

Reply:
thx, resolving